File: src/main/java/ca/warp7/pathplotter/state/ControlPoint.java
package ca.warp7.pathplotter.state;

import edu.wpi.first.wpilibj.geometry.Pose2d;

import java.util.Objects;

public class ControlPoint {
    public Pose2d pose;
    public boolean isSelected = false;
    public double magMultiplier = 1.2;

    public ControlPoint(Pose2d pose) {
        this.pose = pose;
    }

    public ControlPoint(Pose2d pose, boolean isSelected, double magMultiplier) {
        this.pose = pose;
        this.isSelected = isSelected;
        this.magMultiplier = magMultiplier;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ControlPoint controlPoint = (ControlPoint) o;
        return pose.equals(controlPoint.pose);
    }

    @Override
    public int hashCode() {
        return Objects.hash(pose);
    }
}

File: src/main/kotlin/ca/warp7/pathplotter/state/ControlPoint.kt
package ca.warp7.pathplotter.state

import edu.wpi.first.wpilibj.geometry.Pose2d

class ControlPoint(var pose: Pose2d) {
    var isSelected = false


    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ControlPoint

        if (pose != other.pose) return false

        return true
    }

    override fun hashCode(): Int {
        return pose.hashCode()
    }

}